rep-2012.rst

```diff
@@ -0,0 +1,466 @@
+REP: 2012
+Title: Service Introspection
+Author: Aditya Pande <aditya.pande@openrobotics.org>, Brian Chen <brian.chen@openrobotics.org>, Deepanshu Bansal <deepanshu.bansal@openrobotics.org>, Jacob Perron <jacob@openrobotics.org>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 07-Jun-2022
+Post-History:
+
+Abstract
+========
+
+This REP proposes a feature to introspect ROS services during runtime.
+The feature allows users to remotely monitor service requests and responses.
```

Comment:
Since we have not been able to introspect service calls up until this point, there has perhaps been the chance that some ROS2-based project has used services to exchange private information between nodes under the assumption that it cannot be intercepted. This might be a good argument for keeping monitoring opt-in, with a clear warning that this might have security implications.

Reply:
> Since we have not been able to introspect service calls up until this point, there has perhaps been the chance that some ROS2-based project has used services to exchange private information between nodes under the assumption that it cannot be intercepted. This might be a good argument for keeping monitoring opt-in, with a clear warning that this might have security implications.

You could already do that AFAIK, as services were using DDS topics.
The fact that it was not possible from ros2 tools didn't make it secure (security by obscurity is no security).

You should be able to configure encription, authentication and access control using sros2.
The only thing is that it would be nice to publish all events in separated topics from a security point of view (one events topic per service, one events topic for the server and another for the client), so access control can be configured granularly (cc @jacobperron ).

But otherwise I don't think this proposal should affect security, though I'm not an expert.